src/components/Header/TopNav/ServiceLink.js

```diff
@@ -21,26 +21,23 @@ const StyledServiceLink = styled(Link)`
   }
 `
 
-const ServiceLink = ({ path = '/' }) => {
-  const isServicePage = getIsServicePage(path)
-  const serviceTitle = isServicePage ? path.match(servicesRegExp)[0] : null
-
-  const isSpecialityPage = path.includes('speciality')
-  const service = isSpecialityPage ? getService(path) : serviceTitle
-
-  return (
-    <Fragment>
-      {isServicePage || isSpecialityPage ? (
-        <StyledServiceLink
-          to={`/${service}`}
-          title={service}
-          color={isServicePage ? 'text' : 'white'}
-        >
-          {capitalize(service)}
-        </StyledServiceLink>
-      ) : null}
-    </Fragment>
+const getServiceFromSlug = slug =>
+  Object.keys(specialitiesMap).find(speciality =>
+    specialitiesMap[speciality].includes(slug)
   )
+
+const ServiceLink = ({ slug, isServicePage, isSpecialityPage }) => {
+  const service = isServicePage ? slug : getServiceFromSlug(slug)
+
+  return isServicePage || isSpecialityPage ? (
+    <StyledServiceLink
+      to={`/${service}`}
+      title={service}
+      color={isServicePage ? 'text' : 'white'}
+    >
+      {capitalize(service)}
+    </StyledServiceLink>
+  ) : null
 }
```

Comment:
I hate a ternary that has `: null` in React. I know it's legacy code but let's improve:
`(isServicePage || isSpecialityPage) && <StyledServiceLink`

Reply:
I agree with @GBouffard ;)